api/src/repositories/attachment-repository.ts

```diff
@@ -132,6 +145,49 @@ export class AttachmentRepository extends BaseRepository {
     return response.rows[0];
   }
 
+  /**
+   * Query to get all project attachment by the given attachment IDs
+   * @param {number} projectId The ID of the project
+   * @param {number[]} attachmentIds The ID of the attachment
+   * @return {Promise<IProjectAttachment[]>} The project attachment having the given IDs.
+   * @memberof AttachmentRepository
+   */
+  async getProjectAttachmentsByIds(projectId: number, attachmentIds: number[]): Promise<IProjectAttachment[]> {
+    defaultLog.debug({ label: 'getProjectAttachmentsByIds' });
+
+    const knex = getKnex();
+
+    const queryBuilder = knex
+      .queryBuilder()
+      .select([
+        'project_attachment_id AS id',
+        'uuid',
+        'file_name',
+        'file_type',
+        'title',
+        'description',
+        'create_user',
+        'update_date',
+        'create_date',
+        'file_size',
+        'key'
+      ])
+      .from('project_attachment')
+      .whereIn('project_attachment_id', attachmentIds)
+      .andWhere('project_id', projectId);
+
+    const response = await this.connection.knex<IProjectAttachment>(queryBuilder);
+
+    if (!response.rows) {
+      throw new ApiExecuteSQLError('Failed to get project attachments by attachmentIds', [
+        'AttachmentRepository->getProjectAttachmentsByIds',
+        'rows was null or undefined, expected rows != null'
+      ]);
+    }
```

Comment:
This check could be removed because rows will always be an array. That being said, it is in a bunch of other places, so we could opt to clean them all up separately later, as tech debt. Up to you though.